Mathlib/NumberTheory/NumberField/Basic.lean

```diff
@@ -65,34 +65,66 @@ instance [NumberField L] [Algebra K L] : FiniteDimensional K L :=
 
 /-- The ring of integers (or number ring) corresponding to a number field
 is the integral closure of ℤ in the number field. -/
```

Comment:
```suggestion
is the integral closure of ℤ in the number field.

This is defined as its own type, rather than a `Subalgebra`, for performance reasons:
looking for instances of the form `SMul (RingOfIntegers _) (RingOfIntegers _)` makes
much more effective use of the discrimination tree than instances of the form
`SMul (Subtype _) (Subtype _)`.
The drawback is we have to copy over instances manually.
-/
```